static/js/disease/data_processing_utils.ts

```diff
@@ -1,9 +1,56 @@
 import _ from "lodash";
 
 import { GraphNodes } from "../shared/types";
-import { DiseaseGeneAssociationData } from "./chart";
+import { DiseaseGeneAssociationRawData } from "./chart";
 export function getDiseaseGeneAssociation(
   data: GraphNodes
-): DiseaseGeneAssociationData[] {
-  return;
+): DiseaseGeneAssociationRawData[] {
+  // Disease to gene associations
+  // checks if the data is empty
+  if (!data) {
+    return [];
+  }
+  const returnData: DiseaseGeneAssociationRawData[] = [];
+  // checks for null values
+  if (_.isEmpty(data.nodes) || _.isEmpty(data.nodes[0].neighbors)) {
+    return [];
+  }
+  for (const neighbour of data.nodes[0].neighbors) {
+    if (neighbour.property !== "diseaseOntologyID") {
+      continue;
+    }
+    for (const node of neighbour.nodes) {
+      let gene = null;
+      let score = null;
+      let interval = null;
+      // check for null or non-existent property values
+      if (_.isEmpty(node.neighbors)) {
+        continue;
+      }
+      for (const n of node.neighbors) {
+        if (n.property === "geneID") {
+          // check for empty list and null gene values
+          if (_.isEmpty(n.nodes) || _.isEmpty(n.nodes[0].value)) {
```

Comment:
nit: and since you're clever more than once this could also be a good future helper

Reply:
you're right! Since its just one line of code, I wasn't sure if we'd want to make a function though!